### PR TITLE
以beanName查已激活的模块的Bean

### PR DESCRIPTION
**Pull request by gaosaroma**
## Summary by CodeRabbit

- **New Features**
  - Introduced methods for retrieving specific services, enhancing service retrieval and error handling.
  - Added a function for improved service proxy management, allowing targeted service retrieval based on criteria.
  - Implemented a method to validate the existence of modules, increasing robustness.

- **Bug Fixes**
  - Enhanced the testing framework to manage multiple business contexts, improving accuracy in service retrieval and bean management.

- **Tests**
  - Expanded test coverage with new scenarios for module service retrieval, including handling of exceptions and multiple contexts.

### CODE REVIEW

File: koupleless-common/src/main/java/com/alipay/sofa/koupleless/common/service/ServiceProxyFactory.java [hunk 1]
```diff
@@ -92,6 +92,22 @@ public static <T> Map<String, T> batchCreateServiceProxy(String bizName, String
         return proxyMap;
     }
 
+    public static <T> T filterServiceProxy(String bizName, String bizVersion, String name,
```
Comment (lvjing2) [inline koupleless-common/src/main/java/com/alipay/sofa/koupleless/common/service/ServiceProxyFactory.java:95]: 和63行的方法是不是一样的

File: koupleless-common/src/main/java/com/alipay/sofa/koupleless/common/api/SpringServiceFinder.java [hunk 2]
```diff
@@ -91,6 +91,41 @@ public static <T> T getModuleService(String moduleName, String moduleVersion, St
             null);
     }
 
+    /**
+     * 遍历所有模块，找name对应的服务，存在多个则报错，不存在则返回null
+     * @param name a {@link java.lang.String} object
+     * @param serviceType a {@link java.lang.Class} object
+     * @return T
+     */
+    public static <T> T getModuleService(String name, Class<T> serviceType) {
```
[the remark this answers is not in the document]
Comment (gaosaroma) [reply koupleless-common/src/main/java/com/alipay/sofa/koupleless/common/api/SpringServiceFinder.java]: 已修复

File: koupleless-common/src/main/java/com/alipay/sofa/koupleless/common/service/ServiceProxyFactory.java (hunk 1 above)
Comment (gaosaroma) [reply koupleless-common/src/main/java/com/alipay/sofa/koupleless/common/service/ServiceProxyFactory.java:95]: 不太一样。在获取 bean 时，如果该bean不存在，createServiceProxy 会抛出异常；filterServiceProxy 会返回 null

File: koupleless-common/src/main/java/com/alipay/sofa/koupleless/common/api/SpringServiceFinder.java [hunk 3]
```diff
@@ -91,6 +94,75 @@ public static <T> T getModuleService(String moduleName, String moduleVersion, St
...
+    }
+
+    /**
+     * 找指定模块的 name 对应的可用服务
+     * @param moduleName a {@link java.lang.String} object
+     * @param serviceName a {@link java.lang.String} object
+     * @param serviceType a {@link java.lang.Class} object
+     * @return T
+     */
+    public static <T> T getModuleServiceWithoutVersion(String moduleName, String serviceName,
```
Comment (lvjing2) [inline koupleless-common/src/main/java/com/alipay/sofa/koupleless/common/api/SpringServiceFinder.java:140]: 我们很多地方都已经用 bizName了，看下是否保持一致

File: koupleless-common/src/main/java/com/alipay/sofa/koupleless/common/service/ServiceProxyFactory.java [hunk 4]
```diff
@@ -92,6 +92,26 @@ public static <T> Map<String, T> batchCreateServiceProxy(String bizName, String
...
+        Biz biz = ArkClient.getBizManagerService().getBiz(bizName, bizVersion);
+        Class<?> serviceClass = tryToGetTargetClass(bizName, bizVersion, biz, serviceType);
+
+        if (null == serviceClass) {
+            return null;
+        }
+
+        Map<String, ?> serviceMap = listService(biz, serviceClass);
+        for (String beanName : serviceMap.keySet()) {
+            if (beanName.equals(name)) {
```
Comment (lvjing2) [inline koupleless-common/src/main/java/com/alipay/sofa/koupleless/common/service/ServiceProxyFactory.java:106]: 这里一个 beanName 只有一个匹配吧，可以直接使用 serviceMap.get 方法不需要 for 循环吧，

Comment (gaosaroma) [reply koupleless-common/src/main/java/com/alipay/sofa/koupleless/common/service/ServiceProxyFactory.java]: 对。。。 我改好了

**Review by lvjing2 — APPROVED** [review]
LGTM

**Pull request merged**